### ISSUE TITLE
Rule: no-watch 

### ISSUE THREAD

**Issue by binjospookie** [effector/eslint-plugin#48]
Watch have last  [computation priority](https://effector.dev/docs/advanced-guide/computation-priority).

Frequent use of the watch can lead to races that are very difficult to debug. Often, the use of the watch can be avoided (by using `target`, for example).

**Comment by igorkamyshev** [comment]
But effects has the same computation priority 🤔 

I agree, we can forbid `.w because it leads to imperative code, but I cannot find the connection with computation priority.

**Comment by binjospookie** [comment]
> but I cannot find the connection with computation priority

```ts
forward({
  from: fooEvt,
  to: barEvt
});
```

```ts
fooEvt.watch(barEvt);
```

The second option can lead to an "unexpected" calculation order.

**Comment by binjospookie** [comment]
If u want I can try to implement this rule. Both my issues are not "do this for us" :)

**Comment by igorkamyshev** [comment]
So, I agree, it's a good rule. Let's do it. I think we ought to mark this rule as `warn`.

**Comment by binjospookie** [comment]
Maybe u can give me some recommendations how to write new rule? I wrote rules many-many years ago :c

**Comment by igorkamyshev** [comment]
Oh, I don't think so 🥲 my only advice: try to write as many tests as you can. ESLint plug-in documentation is a really poor.

**Comment by binjospookie** [comment]
Need a little help. What we need to do with the JS way branch? `.watch` can be called in a file that does not declare a watchable unit. We can't rely on unit naming, because there is no naming convention for events.

**Comment by igorkamyshev** [comment]
Such a good question. I don't have an answer for you.

Maybe, we ought to introduce https://github.com/effector/eslint-plugin/issues/32 before?

But, it isn't resolving a problem with something like `sample(...).watch(...)` and so on.

So, in my opinion, we have two ways:
1. Specify this rule only for TS-code
2. Don't make this rule

What do you think?

**Comment by binjospookie** [comment]
I like the first way. I'll explain why. 

Don't take the rule away from developers with TypeScript and Effector projects just because not all projects have TypeScript.

Binding to name convention seems fragile because any naming rule can be disabled.

**Comment by igorkamyshev** [comment]
I agree. But we should notice it in documentation.
